### ISSUE TITLE
Support additional default accept headers: application/fhir+json, application/json+fhir

### ISSUE THREAD

**Issue by kolkheang** [smart-on-fhir/client-js#23]
Currently the [default accept header](https://github.com/smart-on-fhir/fhir.js/blob/master/src/fhir.js#L32) for all FHIR requests is `application/json`.  The spec for STU 3 is using `application/fhir+json` and the current DSTU 2 is `application/json+fhir`.  

The library should default in the appropriate accept headers and let each implementation of FHIR server deal with content negotiation properly.  This way the client will get the correct content specified.

This can be done [here](https://github.com/smart-on-fhir/fhir.js/blob/master/src/fhir.js#L32) or during [initialization](https://github.com/smart-on-fhir/client-js/blob/master/src/client/client.js#L41-L43) by passing the accept header values in order.

**Comment by kolkheang** [comment]
The [current fhir.js library](https://github.com/smart-on-fhir/client-js/blob/master/lib/jqFhir.js) being included is not the latest version @ https://github.com/FHIR/fhir.js which supports `Accept` header [override](https://github.com/FHIR/fhir.js/blob/master/src/fhir.js#L32).

**Comment by vlad-ignatov** [comment]
Hi @kolkheang,

v2 is now out, but it still defaults to  `application/json` (although it can be overridden) because it seems to work without having to detect the FHIR version. I was wondering if you have a good real-life example of why  specific headers (like `application/fhir+json`)  would have to be used? Is there a server that you know off, which will ignore `application/json` and only accept `application/fhir+json` or `application/json+fhir`?

**Comment by admah** [comment]
I just ran into a similar issue with the `content-type` header. I'm using v2.2 and trying to create a new entry using DSTU2. With the current `application/fhir+json` header, I get a 415 error. However, when I switch that to `Content-type: application/json+fhir` the request goes through and the new entry is created. 

Has anyone else run into this? Since I can't directly override that header, I'm not sure what else to do. 

On a different note, I'm new to the FHIR space, and have really enjoyed using this library! Thanks to everyone that's worked on it.

**Comment by vlad-ignatov** [comment]
I'll have think about a way to auto-detect this. For now you can pass your headers like so:
```js
client.create(resource, {
    headers: {
        "Content-Type": "application/json+fhir"
    }
})
```

P.S. You will need to update to v2.3.0 for this.

**Comment by admah** [comment]
Thank you! I'll go ahead and try that. You also may not need to. I know Epic supports all MIME-types. Cerner seems to lock them down to a couple of options.

**Comment by admah** [comment]
I'm still seeing the same issue in v2.3.0 specifically related to the `Content-type` header. It looks like the `create` method is still prohibiting that header from being set elsewhere - "Note that `method`, `body` and `headers["Content-Type"]` will be ignored"

**Comment by vlad-ignatov** [comment]
Oh, yes!  I will definitely have to allow that header in the next version.

BTW, since you managed to modify the headers somehow, I wonder if `application/json` works in your case?

**Comment by admah** [comment]
Yes, `application/json` works fine.

**Comment by vlad-ignatov** [comment]
This should be fixed in `2.3.1`. The content type for `create` and `update` defaults to `application/json` and can be overridden.

Note that the library has everything that would be needed to automatically detect and use `application/fhir+json` or `application/json+fhir`. However, I hope that won't be necessary because it is expensive to download and parse the `CapabilityStatement` just to get the FHIR version...
